The library crashes on Android when closeSSEConnection() is called

Hi guys,
The library crashes in closeSSEConnection() call, the stack trace looks like:

Caused by: android.os.NetworkOnMainThreadException
  at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1303)
  at java.net.SocketInputStream.read(SocketInputStream.java:150)
  at java.net.SocketInputStream.read(SocketInputStream.java:120)
  at okio.Okio$2.read(Okio.java:139)
  at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
  at okio.RealBufferedSource.indexOf(RealBufferedSource.java:345)
  at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:217)
  at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:211)
  at okhttp3.internal.http1.Http1Codec$ChunkedSource.readChunkSize(Http1Codec.java:463)
  at okhttp3.internal.http1.Http1Codec$ChunkedSource.read(Http1Codec.java:446)
  at okhttp3.internal.Util.skipAll(Util.java:175)
  at okhttp3.internal.Util.discard(Util.java:157)
  at okhttp3.internal.http1.Http1Codec$ChunkedSource.close(Http1Codec.java:484)
  at okio.RealBufferedSource.close(RealBufferedSource.java:455)
  at com.here.oksse.RealServerSentEvent$Reader.close(RealServerSentEvent.java:199)
  at com.here.oksse.RealServerSentEvent.close(RealServerSentEvent.java:142)
  at xx.xxxxx..controller.EventsController.closeSSEConnection(EventsController.java:140)
 
The source of trouble that at line #199 of RealServerSentEvent.java the method source.close() is called from the main thread. If the library is designed only for Java, not Android, sorry that I raised this issue. Otherwise the library should take care to correctly handle this situation. Also it should send callback events from the main thread, not the Okio execution (background) thread.

--- a/src/main/java/com/here/oksse/RealServerSentEvent.java
+++ b/src/main/java/com/here/oksse/RealServerSentEvent.java
@@ -93,10 +93,7 @@ class RealServerSentEvent implements ServerSentEvent {
     private void notifyFailure(Throwable throwable, Response response) {
         if (!retry(throwable, response)) {
             listener.onClosed(this);
-            if (call != null && !call.isCanceled()) {
-                call.cancel();
-                close();
-            }
+            close();
         }
     }
 
@@ -135,11 +132,8 @@ class RealServerSentEvent implements ServerSentEvent {
 
     @Override
     public void close() {
-        if (call != null) {
+        if (call != null && !call.isCanceled()) {
             call.cancel();
-        }
-        if (sseReader != null) {
-            sseReader.close();
         }
     }
 
@@ -189,17 +183,6 @@ class RealServerSentEvent implements ServerSentEvent {
                 return false;
             }
             return true;
-        }
-
-        /**
-         * Close the source
-         */
-        void close() {
-            try {
-                source.close();
-            } catch (IOException e) {
-                // Close quietly
-            }
         }
 
         /**

